### DISCUSSION TITLE
Change from top to bottom margin?

### DISCUSSION THREAD

**Discussion by jsbrain** [tailwindlabs/tailwindcss#2935] [Help] [answered]
Hey @adamwathan ,

I noticed that on the new v2 website you "switched" from top margin to bottom margin.
In one of your screencasts you say that you prefer top margin but I also saw a twitter post
from you earlier this year where you said you're questioning a lot of things.

So my question: Could you please explain if there's any specific reason on that very site
or if you generally "switched" from preferring top vs bottom margin?

Would be very grateful for just a brief explanation, I'm very curious what led to the change :)

Thanks!

**Comment by MathGeniusJodie** [comment]
https://mxstbr.com/thoughts/margin

**Comment by simonswiss** [comment, marked as answer]
Adam might chime in, but I wouldn't read _too much_ into this - the short answer is: @bradlc build the new website, not Adam 😅

We're trying to standardize our coding style as much as possible across all our products but we're effectively still a fairly "new" team working on different things 👍

**My** personal preference is to apply margin on `top` and `left` only for most situations (in a "left-to-right" scenario), for the following reason:

Each element on a page should be responsible for its own layout. If the incoming element needs vertical spacing between itself and the previous element, it should handle that spacing itself. The previous element should not have to say: 

> "Hey, I don't want anything too close to me, so I am going to put some margin next to me or below me in anticipation for the next thing that comes up."

Instead, I want the current incoming element to say:

> "Hey - I want to have some space between myself and that previous element here, so let me apply some margin op top of me or next to me to handle that spacing"

Not sure it even makes sense explained like this - but it sure makes sense in my head 😅

**Reply by MathGeniusJodie** [reply]
that's really smart

**Reply by jsbrain** [reply]
Hey @simonswiss thank you for your reply and I absolutely agree with that and that's the way I'm doing it since I first
twisted my mind around that. I was just curious if there's any recent stuff going on that changed Adams mind.

But yeah that explains why it is different on the new website, thank you :D
